test/core/store/fs/impl/test_cog.py

```diff
@@ -0,0 +1,193 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import os.path
+import unittest
+
+import fsspec
+import rasterio as rio
+import rioxarray
+import s3fs
+import xarray
+import xarray as xr
+
+# from xcube.core.store.fs.impl import cog
+# from test.s3test import S3Test, MOTO_SERVER_ENDPOINT_URL
+from xcube.core.store.fs.impl.cog import GeoTIFFMultiLevelDataset, \
+    MultiLevelDatasetGeoTiffFsDataAccessor
+from xcube.core.store.fs.impl.dataset import DatasetGeoTiffFsDataAccessor
+from xcube.util.jsonschema import JsonObjectSchema
+
+
+class RioXarrayTest(unittest.TestCase):
+
+    def test_it_is_possible_get_overviews(self):
+        cog_path = 'examples/serve/demo/cog-example.tif'
+        array = rioxarray.open_rasterio(cog_path)
+        self.assertIsInstance(array, xr.DataArray)
+        self.assertEqual((3, 9984, 22016), array.shape)
+
+        rio_accessor = array.rio
+        self.assertIsInstance(rio_accessor,
+                              rioxarray.raster_array.RasterArray)
+        manager = rio_accessor._manager
+        print(manager)
+        self.assertIsInstance(manager,
+                              xr.backends.CachingFileManager)
+        rio_dataset = manager.acquire(needs_lock=False)
+        self.assertIsInstance(rio_dataset,
+                              rio.DatasetReader)
+
+        overviews = rio_dataset.overviews(1)
+        self.assertEqual([2, 4, 8, 16, 32, 64],
+                         overviews)
+
+        num_levels = len(overviews)
+        shapes = []
+        for i in range(num_levels):
+            array = rioxarray.open_rasterio(cog_path,
+                                            overview_level=i)
+            shapes.append(array.shape)
+
+        self.assertEqual([(3, 4992, 11008),
+                          (3, 2496, 5504),
+                          (3, 1248, 2752),
+                          (3, 624, 1376),
+                          (3, 312, 688),
+                          (3, 156, 344)],
+                         shapes)
+
+        with self.assertRaises(rio.errors.RasterioIOError) as e:
+            rioxarray.open_rasterio(cog_path,
+                                    overview_level=num_levels)
+        self.assertEqual(
+            (
+                'Cannot open overview level 6 of '
+                'examples/serve/demo/cog-example.tif',),
+            e.exception.args
+        )
+
+
+class GeoTIFFMultiLevelDatasetTest(unittest.TestCase):
+
+    def test_local_fs(self):
+        fs = fsspec.filesystem('file')
+        cog_path = "examples/serve/demo/cog-example.tif"
```

Comment:
1. Use `__file__` to make the test independent of the current working directory
2. "cog-example.tif" is too large, please remove it and use the other one.

Reply:
```suggestion
        cog_path = os.path.join(os.path.dirname(__file__),
                                "..", "..", "..", "..", "..",
                                "examples", "serve", "demo",
                                cog_filename)
```